kb/python-sandbox-environmnet.md

```diff
@@ -0,0 +1,132 @@
+---
+slug: python-sandbox-environment
+title: Python sandboxed-environments
+tags:
+  - kb-article
+date: 2022-11-15T00:00:00Z
+---
+
+The Temporal Python SDK allows you to run Workflow code in a sandboxed-environment to help prevent non-determinism errors in your application. The Temporal Workflow Sandbox for Python is not completely isolated, and some libraries can internally mutate state, which can result in breaking determinism.
+
+<!-- truncate -->
+
+By default, Workflows run in a sandboxed-environment. If a Workflow Execution performs a non-deterministic event, an exception is thrown, which results in failing the Task Worker. The Workflow will not progress until the code is fixed.
+
+## Benefits
+
+Temporal's Python SDK uses a sandboxed-environment for Worfklow runs to make developing Workflow code safer, easier, and its cost on performance is relatively little.
+
+## How it works
+
+The Sandbox environment is made up of two main components.
+
+- [Global state isolation](#global-state-isolation)
+- [Restrictions](#restrictions)
+
+### Global state isolation
+
+The first component of the Sandbox is a global state isolation. Global state isolation uses `exec` to compile and evaluate statements.
+
+Upon the start of a Workflow, the file that the Workflow is defined in, is imported into a newly created sandbox.
+
+A known set of pass-through modules are sent to the sandbox when the Workflow is imported.
```

Comment:
This isn't really accurate. Rather if a module is imported by the file, a known set (which includes all of Python stdlib) is "passed through" from outside the sandbox.